language/libra-vm/src/lib.rs

```diff
@@ -150,3 +155,9 @@ pub trait VMExecutor: Send {
         state_view: &dyn StateView,
     ) -> Result<Vec<TransactionOutput>, VMStatus>;
 }
+
+/// Get the AccessPath to a resource stored under `address` with type name `tag`
+fn create_access_path(address: AccountAddress, tag: StructTag) -> AccessPath {
```

Comment:
is the caching of the resource_tag not around anymore?
that seems pretty vital to me but I may have not seen it

Reply:
Not implemented yet. I have that planned but will do some benchmarks without caching to help us confirm the necessity of it.

Reply:
So you can imagine there are type additional caches I may implement:
1. type => type tag, type layout
  - This is in the Move VM.
2. type tag => AccessPath
  - This is in the Libra VM.

Reply:
I think the cache may have decent implications to be honest and try to get to that asap.
We had to add it back because of perf related reasons so I don't see how it's not going to matter any longer. It was not just an optimization for the sake of it.
So let's get that info asap or we may hit surprises.
I hoped we could remove caches from the loader that are not related to runtime. Of course the loader would have to be used for resolution of certain type information.
Don't know we could go either way I guess.

Let's add the caches asap though

Reply:
Ah my bad I didn't make this clear. I'll definitely add the cache implementation to this PR before it lands. It's just that I wanna run benchmarks without them first so I can compare the two. I absolutely trust your judgement on the importance of the caches, just saying some first hand experiments/data will help me understand this even better.